draft.md

```diff
@@ -116,10 +116,10 @@ API publishers SHOULD use following values for the field:
   - "warn": healthy, with some concerns.
 
   The value of the status field is case-insensitive and is tightly related with
-  the HTTP response code returned by the health endpoint. For “pass” and “warn”
-  statuses, HTTP response code in the 2xx-3xx range MUST be used. For “fail”
-  status, HTTP response code in the 4xx-5xx range MUST be used. In case of the
-  “warn” status, endpoints SHOULD return HTTP status in the 2xx-3xx range, and
+  the HTTP response code returned by the health endpoint. For “pass” status,
+  HTTP response code in the 2xx-3xx range MUST be used. For “fail” status,
+  HTTP response code in the 4xx-5xx range MUST be used. In case of the “warn”
```

Comment:
For `fail`, returning a 5XX sounds like a paradox, doesn't it? The endpoint is working and responding with information so it should be an OK code (but it's telling you the service is unavailable) hehe. Also, I'm not sure, but isn't there a risk of having "response bodies with status information" dropped because of using a 5XX in some systems out there? What's your view?

Reply:
Relying on status codes only and entirely ignoring message bodies is already a well-established pattern in cloud-native autonomous control plane systems.

I’ll dig up some Kubernetes and Azure links where they talk about relying only on status codes from health/liveness/readiness endpoints for orchestration concerns.

Reply:
It does indeed make more sense to use a 4xx error code when an API can respond and it is in 'fail' mode. However, if it cannot even respond, the response code will be a 5xx so 5xx code is equivalent of 'fail'. To ensure there are no weird edge-cases it therefore follows that responding with a 5xx for some 'fails' should be ok. :) /shrug